### DISCUSSION TITLE
Bug in the Mirror!!!

### DISCUSSION THREAD

**Discussion by Vanillma** [termux/termux-packages#17756] [Q&A] [answered]
Hey guys.
It has been a few days that one of the mirrors has had a problem.
This mirror is located in Iran and was created by a person named Berdia.
I tried to contact him and inform him...but he says the problem is with the internet providers in Iran...while the error text clearly complains about something else.
I also tested with VPN and got the same error again... If the problem is from the internet provider, it should have been solved with VPN... but it was not.
Seems to be due to misconfiguration of the mirror...what do you think??
This is the error text:
```
Get:1 https://mirror.bardia.tech/termux/termux-main stable InRelease [14.0 kB]
Get:2 https://mirror.bardia.tech/termux/termux-main stable/main arm Packages [487 kB]
Err:2 https://mirror.bardia.tech/termux/termux-main stable/main arm Packages
  File has unexpected size (486202 != 486771). Mirror sync in progress? [IP: 185.142.159.242 443]
  Hashes of expected file:                           - Filesize:486771 [weak]
   - SHA512:000d1fa5f28b2ca2fbbfde4b03517d994f6ae5c16447063ed18434f66eea3a545f76460ca74fc35393d9f865e88bf4096e4bb0a1d5b2ddec03837e75cb2fb6f7
   - SHA256:ae2d61b56e1944ff6ea50d97bd8c408f86b0a1017fd6d0acee42a62f7f2dd093
   - SHA1:6236c41f6d12d38b8689b9d0b42c66367b2ac531 [weak]
   - MD5Sum:2eebfc99cac7285cd0ef2e406756d65e [weak]
  Release file created at: Mon, 04 Sep 2023 00:51:42 +0000
Fetched 14.0 kB in 5s (2988 B/s)
Reading package lists... Done
E: Failed to fetch https://mirror.bardia.tech/termux/termux-main/dists/stable/main/binary-arm/Packages.bz2  File has unexpected size (486202 != 486771). Mirror sync in progress? [IP: 185.142.159.242 443]
   Hashes of expected file:
    - Filesize:486771 [weak]
    - SHA512:000d1fa5f28b2ca2fbbfde4b03517d994f6ae5c16447063ed18434f66eea3a545f76460ca74fc35393d9f865e88bf4096e4bb0a1d5b2ddec03837e75cb2fb6f7
    - SHA256:ae2d61b56e1944ff6ea50d97bd8c408f86b0a1017fd6d0acee42a62f7f2dd093
    - SHA1:6236c41f6d12d38b8689b9d0b42c66367b2ac531 [weak]
    - MD5Sum:2eebfc99cac7285cd0ef2e406756d65e [weak]
   Release file created at: Mon, 04 Sep 2023 00:51:42 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by sylirre** [comment, marked as answer]
> I also tested with VPN and got the same error again... If the problem is from the internet provider, it should have been solved with VPN...

No, it shouldn't. If the mirror is located in Iran where Internet censorship takes place, the mirror server has limited access to the origin server. This means it is unable to do the synchronization.

Using VPN from your side won't solve the issue.

> Seems to be due to misconfiguration of the mirror...

Again, no. Look at the error message: it is about file integrity check failure.

Apt repository provides the package files (.deb files) and metadata files which are of two types:

* The package lists which provide information about packages including their size and checksums. These files have names like `Packages.bz2`, `Packages.gz` or just `Packages`.
* Release information such as repository update date, checksums of the package lists and cryptographic signature. These files named `InRelease` (embedded signature) or `Release` with signature in separate file `Release.gpg`.

In your case the package list is out of sync with release information. Basically means that repository is in inconsistent state (e.g. the connection broke during synchronization). It can be fixed only by full synchronization with the origin server.

**The best thing you can do is TO NOT USE this mirror.** Use `termux-change-repo`, select single mirror mode and then choose a different one (controls: space to select, enter to confirm selection).

*Most of the mirrors are operated by third party persons which are not affiliated with @termux.*

**Reply by Vanillma** [reply]
Thank you very much for the explanation....Yes, I know they belong to a third party, But the file size is unexpected, it made me suspect that the mirroring process might have a problem!!
With your explanation, I now understand that it is because of censorship!

**Reply by sylirre** [reply]
Actually both. The intrusion into network connection caused a problem with downloading changes. As result the mirror became inconsistent and unusable.

Next successful sync should fix it. For now just use other mirror that works for you.

**Reply by Vanillma** [reply]
Yes you were right...looks like it's fixed now!
I thank you for the complete explanation And also for the time you took to reply.
![Screenshot_20230904-220850](https://github.com/termux/termux-packages/assets/95992420/a8ced36f-4b23-4756-964e-cb85c2d3ff91)

**Comment by ArcusReborne** [comment]
I know this is old but I just wanted to mention this for those that come along like me, I solved a similar issue by running:

```
apt clean
pkg clean
rm -rf /data/data/com.termux/files/usr/var/lib/apt/lists/*

apt upgrade
pkg upgrade

```
My issue was a package desync, so the idea is to clear files and redownload them.
